src/include/distributed/pg_dist_partition.h

```diff
@@ -54,6 +54,10 @@ typedef FormData_pg_dist_partition *Form_pg_dist_partition;
 #define DISTRIBUTE_BY_RANGE 'r'
 #define DISTRIBUTE_BY_NONE 'n'
 #define REDISTRIBUTE_BY_HASH 'x'
+#define CITUS_LOCAL_TABLE 'l'
+
+#define CitusTableWithoutDistributionKey(x) \
+	((x) == DISTRIBUTE_BY_NONE || (x) == CITUS_LOCAL_TABLE)
```

Comment:
I have been wondering: Should this be a 'none' distributed table as well and should there be a separate column in pg_dist_partition for how the table is replicated? The semantics of "distribute by" is gradually getting more awkward.

Reply:
> Should this be a 'none' distributed table as well 

That sounds good to me, and actually makes a lot of sense.


>  a separate column in pg_dist_partition for how the table is replicated?

we already have `pg_dist_partition.repmodel`. Can we use that, or do you have anything else in mind? 

We mark `t` (`REPLICATION_MODEL_2PC`) for  reference tables, and we simply use `citus.replication_model` for  coordinator local tables -- which can never be `REPLICATION_MODEL_2PC`. So, it seems we have enough information in the metadata distinguish `reference table` vs `citus local` by marking both `DISTRIBUTE_BY_NONE `.

Reply:
Actually I explicitly didn't want to name this macro as "IsNoneDistributedTable" as it seems more like describing the reference tables.

The second approach makes more sense to me, like adding another column to `pg_dist_partition` and not using `PartitionMethod` for reference tables and citus local tables anymore.

But I guess, it would require more changes, maybe after merging citus local table pr's I can do such a refactor ?

Reply:
>But I guess, it would require more changes, maybe after merging citus local table pr's I can do such a refactor ?

I don't think that should be a refactor after merging this to the master. That seems like a fundamental change, that we should do before merging into master.

 > like adding another column to pg_dist_partition

Do you have any reservations on using `pg_dist_partition.repmodel` ?

Reply:
Normally, it seems that we do not manually set `replicationModel` for citus tables when creating them, instead we call `AppropriateReplicationModel` at the very beginning of `CreateDistributedTable` and `AppropriateReplicationModel` relies on `partitionMethod`.

I wonder in that case, shouldn't we call this function for citus local tables if we remove `partitionMethod` enum for citus local tables ? We pass another parameter to `CreateDistributedTable` and we won't be calling `AppropriateReplicationModel` if `replicationModel` parameter given is valid & and equal to `REPLICATION_CITUS_LOCAL` (or something like this)

Also, the way we check if a table has no distribution column is very simple now, we just call the macro I wrote `CitusTableWithoutDistributionKey` which only requires a single parameter